source/ota_mqtt.c

```diff
@@ -282,6 +282,8 @@ static size_t stringBuilder( char * pBuffer,
         curLen += thisLength;
```

Comment:
Beyond the scope of this PR, but I dislike that we are only asserting (on line 279) on the string length not exceeding the buffer size. In case assert is not defined, we'll simply cause memory corruption.

Reply:
Is there a standard way we avoid this? I'm happy to incorporate that change.